### PR TITLE
Routers search filter

### PR DESCRIPTION
**Pull request by AgnieszkaPlatek**
Added nested routing, basic search with dynamic fields and ordering by date.

### CODE REVIEW

File: flashcards/helpers.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import filters
+
+
+class CustomSearchFilter(filters.SearchFilter):
+    """
+    Enables the user to specify search fields and if none are chosen both fields are used.
+    """
+    def get_search_fields(self, view, request):
+        return request.GET.getlist('search_fields', ['front', 'back'])
```
Comment (FilipKDick) [inline flashcards/helpers.py]: missing newline

File: flashcards/serializers.py [hunk 2]
```diff
@@ -5,13 +5,15 @@
...
     owner_name = serializers.CharField(read_only=True)
+
     class Meta:
         model = Flashcard
         fields = ['id', 'flashcard_set', 'front', 'back', 'owner', 'owner_name', 'added', 'last_modified']
 
 
 class FlashcardSetSerializer(serializers.ModelSerializer):
-    flashcards = FlashcardSerializer(many=True, read_only=True)
+    # flashcards = FlashcardSerializer(many=True, read_only=True)
```
Comment (FilipKDick) [inline flashcards/serializers.py]: You can remove this line altogether

File: flashcards/urls.py [hunk 3]
```diff
@@ -1,14 +1,19 @@
 from django.urls import path, include
-from rest_framework.routers import DefaultRouter
+from rest_framework_nested import routers
```
Comment (FilipKDick) [inline flashcards/urls.py:4]: Why `nested`?

File: flashcards/helpers.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import filters
+
+
+class CustomSearchFilter(filters.SearchFilter):
```
Comment (FilipKDick) [inline flashcards/helpers.py]: I'd suggest a more detailed name. Right now it's the only custom filter you have, but if you need more, will they all be `CustomSearchFilter` or will this one be an exception?
Rename to `CardFrontBackSearchFilter` or sth

File: flashcards/urls.py (hunk 3 above)
Comment (AgnieszkaPlatek) [reply flashcards/urls.py:4]: I wanted an easy access through url to listing flashcards from the specific set and searching in the specific set. 
An endpoint with all user's flashcards in one place is needed only for searching (for now ;) ), I added it separately.
But if it is not a good idea I can change it back of course.

**Review by FilipKDick — CHANGES_REQUESTED** [review]
FIX IT

**Review by FilipKDick — APPROVED** [review]

**Pull request closed without merging**